SPECS/glusterfs/glusterfs.spec

```diff
@@ -74,74 +81,116 @@
-%if ( 0%{?_tmpfilesdir:1} )
+%if 0%{?_tmpfilesdir:1}
 %global _with_tmpfilesdir --with-tmpfilesdir=%{_tmpfilesdir}
 %else
 %global _with_tmpfilesdir --without-tmpfilesdir
 %endif
 # without server should also disable some server-only components
-%if ( 0%{?_without_server:1} )
+%if 0%{?_without_server:1}
 %global _without_events --disable-events
 %global _without_georeplication --disable-georeplication
+%global _without_linux_io_uring --disable-linux-io_uring
 %global _with_gnfs %{nil}
 %global _without_ocf --without-ocf
 %endif
+%global _usepython3 1
 %global _pythonver 3
-%global service_start()   /bin/systemctl --quiet start %{1}.service || : \
+# From https://fedoraproject.org/wiki/Packaging:Python#Macros
+%if ( 0%{?_with_systemd:1} )
+%global service_start()   %{_bindir}/systemctl --quiet start %1.service || : \
 %{nil}
-%global service_stop()    /bin/systemctl --quiet stop %{1}.service || :\
+%global service_stop()    %{_bindir}/systemctl --quiet stop %1.service || :\
 %{nil}
-%global service_install() install -D -p -m 0644 %{_sourcedir}/%{1}.service %{buildroot}%{2} \
+%global service_install() install -D -p -m 0644 %1.service %{buildroot}%2 \
 %{nil}
 # can't seem to make a generic macro that works
 %global glusterd_svcfile   %{_unitdir}/glusterd.service
 %global glusterfsd_svcfile %{_unitdir}/glusterfsd.service
 %global glusterta_svcfile %{_unitdir}/gluster-ta-volume.service
 %global glustereventsd_svcfile %{_unitdir}/glustereventsd.service
 %global glusterfssharedstorage_svcfile %{_unitdir}/glusterfssharedstorage.service
+%else
+%global systemd_post()  %{_sbindir}/chkconfig --add %1 >/dev/null 2>&1 || : \
+%{nil}
+%global systemd_preun() %{_sbindir}/chkconfig --del %1 >/dev/null 2>&1 || : \
+%{nil}
+%global systemd_postun_with_restart() %{_sbindir}/service %1 condrestart >/dev/null 2>&1 || : \
+%{nil}
+%global service_start()   %{_sbindir}/service %1 start >/dev/null 2>&1 || : \
+%{nil}
+%global service_stop()    %{_sbindir}/service %1 stop >/dev/null 2>&1 || : \
+%{nil}
+%global service_install() install -D -p -m 0755 %1.init %{buildroot}%2 \
+%{nil}
+# can't seem to make a generic macro that works
+%global glusterd_svcfile   %{_sysconfdir}/init.d/glusterd
+%global glusterfsd_svcfile %{_sysconfdir}/init.d/glusterfsd
+%global glustereventsd_svcfile %{_sysconfdir}/init.d/glustereventsd
+%endif
 %{!?_pkgdocdir: %global _pkgdocdir %{_docdir}/%{name}-%{version}}
 # modern rpm and current Fedora do not generate requires when the
 # provides are filtered
 %global __provides_exclude_from ^%{_libdir}/glusterfs/%{version}/.*$
+%global bashcompdir %(pkg-config --variable=completionsdir bash-completion 2>/dev/null)
+%if "%{bashcompdir}" == ""
+%global bashcompdir ${sysconfdir}/bash_completion.d
+%endif
+# Skip regression-tests because they previously were excluded in
+# CBL-Mariner glusterfs and the new version introduces a dependency
+# (dbench)
+%global _without_regression_tests true
+# Explicitly require rpcgen
+%global _require_rpcgen true
```

Comment:
Why do we need this macro? I think we can just leave the parts of the spec it's guarding as enabled and drop it. Especially since we already had `BuildRequires: rpcgen` for a while.

Reply:
ok ... just makes even more departure from the source spec.